text/0249-v2-experiments.md

```diff
@@ -373,6 +379,329 @@ Disadvantages:
+- Low adoption. Users might be hesitant to use an API with a  `PreX` in its name,
+  assuming it means "will break".
+- Cluttering the code base. Although most IDEs will mark deprecated properties
+  with a ~~strikethrough~~, they will still be listed by autocomplete.
+- Will force some pretty long and ugly names on our APIs. Many previews APIs
+  will result in a less aesthetic user code.
+- A lot of deprecated code in aws-cdk-lib, possibly blowing up `aws-cdk-lib`.
+  This might not be a real concern as we can reuse a lot the code between
+  different version of an API.
+
+**How can we encourage users to upgrade to a newer version?**
+
+When a new minor version introduces a breaking change to an API, users have a
+clear motivation to upgrade to the new version of the API - they must do so in
+order to upgrade to a newer version of the AWS CDK. If the API is only
+deprecated, users have no motivation to upgrade to a newer version of the API,
+even worse, they might not be aware that a newer version exists. One way to
+encourage users to upgrade to a newer version of an API, is to supply tools that
+will inform users. For example, we can add a capability to the `cdk doctor`
+command that will notify users that their CDK application is using deprecated
+experimental APIs.
+
+Executing `cdk doctor` will print:
+
+```bash
+neta@dev/my-cdk-application$ cdk doctor
+
+Newer versions of APIs previews your application is using are available. You should consider upgrading.
+To see which previews APIs can be upgraded, execute `cdk --show-deprecated-api-usage`
+```
+
+**If there are no breaking changes, are the APIs really experimental?**
+
+Given that preview APIs are safe to use, and no breaking changes will be
+introduced to them, we might consider not referring to them in any special way,
+and if needed, simply add a version to the API name. The first version of
+`grantWrite` will be named `grantWrite`, the second version will be named
+`grantWriteV1` and so on. While a preview API will not break, we should still
+use a naming scheme that convey its non-final nature for the following reasons:
+
+1. **Real-estate**: `grantWrite` is a much better name than `grantWritePre3` for
+   the final API.
+2. **Encourage feedback**: Declaring an API as "in preview" encourage the
+   community to supply feedback.
+3. **Setting the right expectations**: When an API in preview users are aware
+   that this is not the final version of the API, and its deprecation is
+   expected.
+
+### Modules lifecycle
+
+This section discuss adding a new modules to `aws-cdk-lib`.
+
+#### Alpha (optional)
+
+This stage is intended for modules that are in early development stage, they
+change rapidly and may incurs many breaking changes. These modules will be
+released separately from `aws-cdk-lib` and will have their own version, which
+allows for breaking changes in accordance to semver, e.g `0.x` or prerelease
+qualifiers.
```

Comment:
Should we note here that the decision on how to release those modules is a decision between options 2 and 3?

Seems like this paragraph is suggesting we just chose option 3, but it's not obvious to me why.

Maybe we can defer that decision to later? At the time of reviewing the doc that contains all of the items marked in this PR as ToDo?

Reply:
We need to decide, I'll add a section detailing why I think option 6 is better

Reply:
Add a section which discuss this, I have a strong preference to releasing each module separately. See [multiple-alpha-modules-vs-single-module](https://github.com/aws/aws-cdk-rfcs/blob/neta/option-6/text/0249-v2-experiments.md#multiple-alpha-modules-vs-single-module)